doc/md/fundamentals/2-types/3-functions.md

```diff
@@ -4,25 +4,25 @@ sidebar_position: 3
+:::
 
-Functions can be synchronous or asynchronous. A synchronous function blocks the caller until the function returns with its result, allowing the caller to proceed. This is the usual notion of function implemented by other programming languages.  
+Functions can be synchronous or asynchronous. A synchronous function blocks the caller until it returns with a result, just like functions in most traditional programming languages.
 
-An asynchronous functions immediately returns to its caller. Its execution is decoupled from the execution of the caller. Instead, the caller may receive a **future** value, which acts as a placeholder for the result of the call. This future is replaced with a result when the function call returns. 
+An asynchronous function returns immediately, providing a **future** value as a placeholder for its result. The caller can await the future later to retrieve the result or ignore it and continue with other tasks.
 
-The caller can retrieve the result by awaiting the future at a later point in time. It can also ignore the result entirely by never awaiting the future. Between calling the asynchronous function and awaiting its result, the caller is able to proceed with other tasks.  
+Motoko has functions with different capabilities.
+Local functions, declared using just the `func` keyword, are typically synchronous but can be asynchronous if their body is an `async` expression.
 
-Motoko has different sorts of functions with different capabilities.  
-Local functions, declared using just the `func` keyword, are typically synchronous but can be asynchronous if their body is an `async` expression.  
+Shared functions, declared in actor bodies using the `shared`, `shared query` or `shared composite query` keywords, are asynchronous.
+Calling a shared function sends a message to another actor. The caller of a shared function is typically another Motoko actor, ICP canister or frontend code interacting with the ICP using an agent.
 
-Shared functions, declared in actor bodies using the `shared`, `shared query` or `shared composite query` keywords, are asynchronous. 
-Calling a shared function sends a message to another actor. The caller of a shared function is typically another Motoko actor, ICP canister or frontend code interacting with the ICP using an agent.  
+An actor's shared functions are always called as the result of the actor receiving some message. Shared functions that return a result have `async` return types.
 
-An actor's shared functions are always called as the result of the actor receiving some message. Shared functions that return a result have async return types.  
-
-Since actors can only have `shared` functions as public members, the `shared` keyword can, and usually is, omitted from public actor functions.  
+Since actors can only have `shared` functions as public members, the `shared` keyword can, and usually is, omitted from public actor functions.
```

Comment:
```suggestion
Since actors can only have `shared` functions as public members, the `shared` keyword  is optional and can be omitted from public actor functions.
```